lib/logging.dart

```diff
@@ -364,3 +366,187 @@ class LogRecord {
   @override
   String toString() => '[${level.name}] $loggerName: $message';
 }
+
+typedef _ServiceExtensionCallback = Future<Map<String, dynamic>> Function(
+    Map<String, String> parameters);
+
+/// A shared manager instance whose recorded messages are logged to the
+/// developer console.
+final LogManager logManager = LogManager()..onRecord.listen((record) {
```

Comment:
Maybe name this class by what it does.
DeveloperConsoleLogger, FilteredDeveloperConsoleLogger, etc?